### PR TITLE
ScalametaParser: change like-lambda param parsing

### PR DESCRIPTION
**Pull request by kitbellew**
- change convertToParam, to make sure that type is only parsed in one place, and that not just `using` is allowed as a possible Mod
- also, return Option[] in convertToParam, to avoid checking that it looks like lambda before parsing it as lamda parameter
For #2774.

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -1732,41 +1700,66 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
...
     }
   }
 
-  private def convertToParamClause(tree: Term): Term.ParamClause = (tree match {
-    case Term.Tuple(ts) => ts.flatMap(convertToParam)
-    case _ => convertToParam(tree).toList
-  }).reduceWith(toParamClause(None))
+  private def convertToParamClause(tree: Term)(
+      okNameLike: => Boolean,
+      okParamLike: => Boolean
```
Comment (tgodzik) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1737]:
```suggestion
      isNameLike: => Boolean,
      isParamLike: => Boolean
```
?

**Review by tgodzik — APPROVED** [review]
Super minor comment, otherwise LGTM!

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala (hunk 1 above)
Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1737]: i think the meaning here is not "is this a name" but "is a name allowed here". should i rename to `isNameAllowed"?

Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1737]: I think that would be clearer, thanks!

Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: done

Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: Thanks!

**Pull request merged**